python/seldon_core/seldon_client.py

```diff
@@ -1380,18 +1392,34 @@ def grpc_predict_gateway(deployment_name: str, namespace: str = None, gateway_en
     if channel_credentials is None:
         channel = grpc.insecure_channel(gateway_endpoint, options)
     else:
-        grpc_channel_credentials = grpc.ssl_channel_credentials(root_certificates=open(channel_credentials.certificate_chain_file, 'rb').read(),
-                                                                private_key=open(channel_credentials.private_key_file, 'rb').read(),
-                                                                certificate_chain=open(channel_credentials.root_certificates_file, 'rb').read())
+        # If one of root cert & cert chain are provided, both must be provided
+        #   otherwise there is a null pointer exception in the Go underlying impl
+        if (channel_credentials.private_key_file
+                and channel_credentials.root_certificates_file
+                and channel_credentials.certificate_chain_file):
+            grpc_channel_credentials = grpc.ssl_channel_credentials(
+                root_certificates=open(channel_credentials.root_certificates_file, 'rb').read(),
+                private_key=open(channel_credentials.private_key_file, 'rb').read(),
+                certificate_chain=open(channel_credentials.certificate_chain_file, 'rb').read())
+        # For most usecases only providing the root cert file is enough
+        elif channel_credentials.root_certificates_file:
+            grpc_channel_credentials = grpc.ssl_channel_credentials(
+                root_certificates=open(channel_credentials.root_certificates_file, 'rb').read())
+        # This piece also allows for blank SSL Channel credentials in case this is required
+        else:
+            grpc_channel_credentials = grpc.ssl_channel_credentials()
```

Comment:
The main reason was due to the open statement that is embedded (And yeah it was already in that format). I do agree, I think we'll have to review the whole GRPC / REST SSL at some point soon so not sure how much worth it would be to have it here. 

In regards ot the second question, open files get closed and garbage collected when they lose the last reference pointer, so in this case they would be getting closed either immediately or when the function returns.